crates/subspace-farmer/src/bin/subspace-farmer/commands/farm.rs

```diff
@@ -142,14 +141,20 @@ where
         piece_cache.clone(),
     ));
 
-    let (run_cache_sync_tx, run_cache_sync_rx) = mpsc::channel::<SegmentIndex>(1);
+    let last_segment_index = node_client
+        .farmer_app_info()
+        .await
+        .map_err(|err| anyhow::anyhow!("Couldn't receive history size: {}", err,))?
+        .protocol_info
+        .history_size
+        .segment_index();
```

Comment:
`farmer_app_info` is already retrieved on line 89 above, no need to make another RPC query:
```suggestion
    let last_segment_index = farmer_app_info
        .protocol_info
        .history_size
        .segment_index();
```